src/coreclr/nativeaot/System.Private.CoreLib/src/System/Threading/Thread.NativeAot.cs

```diff
@@ -367,9 +368,13 @@ private void StartCore()
 
                 if (GetThreadStateBit(ThreadState.Unstarted))
                 {
-                    // Lack of memory is the only expected reason for thread creation failure
```

Comment:
This was incorrect assumption. COM initialization can fail with PlatformNotSupportedException on Nano server.